distributed-calculator/README.md

```diff
@@ -38,80 +38,275 @@ Clone the quickstarts repository
-      ```
-      pip3 install wheel
-      pip3 install python-dotenv
-      ```
-    - Set environment variable to use non-default app port 5000
-      ```
-      Linux/Mac OS:
-      export FLASK_RUN_PORT=5000
-
-      Windows:
-      set FLASK_RUN_PORT=5000
-      ```
-    - Start dapr using the command:
-      ```
-      dapr run --app-id multiplyapp --app-port 5000 --dapr-http-port 3501 flask run
-      ```
+
+<!-- STEP
+name: "Build ptyhon app"
+working_dir: "./python"
+-->
+
+- Install required packages
+   ```bash
+   pip3 install wheel python-dotenv flask_cors flask
+   ```
+
+<!-- END_STEP -->
+
+- Set environment variable to use non-default app port 5000
+   ```bash
+   #Linux/Mac OS:
+   export FLASK_RUN_PORT=5000
+   
+   #Windows:
+   set FLASK_RUN_PORT=5000
+   ```
+
+<!-- STEP
+expected_stdout_lines:
+  - "✅  You're up and running! Both Dapr and your app logs will appear here."
+  - "== APP == Calculating 52.0 * 34.0"
+  - "✅  Exited Dapr successfully"
+  - "✅  Exited App successfully"
+expected_stderr_lines:
+name: "Run python app"
+working_dir: "./python"
+background: true
+env:
+  FLASK_RUN_PORT: "5000"
+sleep: 2
+-->
+
+- Start dapr using the command:
+   ```bash
+   dapr run --app-id multiplyapp --app-port 5000 --dapr-http-port 3501 flask run
+   ```
+
+<!-- END_STEP -->
+
 5. Frontend Calculator app - Open a terminal window and navigate to the react-calculator directory and follow the steps below:
-    - Install the required modules
-      ```
-      npm install
-      npm run buildclient
-      ```
-    - Start Dapr using command below:
-      ```
-      dapr run --app-id frontendapp --app-port 8080 --dapr-http-port 3500 node server.js
-      ```
+
+
+<!-- STEP
+name: "Build frontend app"
+working_dir: "./react-calculator"
+timeout_seconds: 600
+-->
+
+- Install the required modules
+   ```bash
+   npm install
+   npm run buildclient
+   ```
+
+<!-- END_STEP -->
+
+
+<!-- STEP
+expected_stdout_lines:
+  - "✅  You're up and running! Both Dapr and your app logs will appear here."
+  - "✅  Exited Dapr successfully"
+  - "✅  Exited App successfully"
+expected_stderr_lines:
+name: "Run frontent app"
+working_dir: "./react-calculator"
+background: true
+sleep: 15
+-->
+
+- Start Dapr using command below:
+   ```bash
+   dapr run --app-id frontendapp --app-port 8080 --dapr-http-port 3500 node server.js
+   ```
+
+<!-- END_STEP -->
+
+
 6. Open a browser window and go to http://localhost:8080/. From here, you can enter the different operations.
 
     ![Calculator Screenshot](./img/calculator-screenshot.JPG)
 
 7. Open your browser's console window (using F12 key) to see the logs produced as you use the calculator. Note that each time you click a button, you see logs that indicate state persistence and the different apps that are contacted to perform the operation. 
 
+8. Curl Validate
+
+
+<!-- STEP
+manual_pause_message: "Calculator APP running on http://localhost:8080. Please open in your browser and test manually."
```

Comment:
This does not need to be manual if the curl commands can be invoked here, right?

Reply:
It doesn't but, the mm.py utility won't actually pause unless it is run with ```--manual```. By default the utility does not pause for manual steps.

Reply:
(I'm still working on a user guide for mm.py, I'll be sure this behavior makes it in)

Reply:
My question is not to pause, but to avoid the manual validation. Can this step run the cURL commands automatically here?

Reply:
The curl commands will be run regardless. The only thing manual_pause_message does is print out this message and wait for the user to type 'x' before continuing if the script is called with ```mm.py --manual README.md```:

```bash
...
Running shell 'bash -c' with command: `dapr run --app-id frontendapp --app-port 8080 --dapr-http-port 3500 node server.js`
Calculator APP running on http://localhost:8080. Please open in your browser and test manually.
Type 'x' to exit
x
Running shell 'bash -c' with command: `curl -w "\n" -s 'http://localhost:8080/calculate/add' -H 'Content-Type: application/json' --data '{"operandOne":"56","operandTwo":"3"}'
 curl -w "\n" -s 'http://localhost:8080/calculate/subtract' -H 'Content-Type: application/json' --data '{"operandOne":"52","operandTwo":"34"}'
 curl -w "\n" -s 'http://localhost:8080/calculate/divide' -H 'Content-Type: application/json' --data '{"operandOne":"144","operandTwo":"12"}'
 curl -w "\n" -s 'http://localhost:8080/calculate/multiply' -H 'Content-Type: application/json' --data '{"operandOne":"52","operandTwo":"34"}'
 curl -w "\n" -s 'http://localhost:8080/persist' -H 'Content-Type: application/json' --data '[{"key":"calculatorState","value":{"total":"54","next":null,"operation":null}}]'
 curl -s 'http://localhost:8080/state' | python -m json.tool`
```

Otherwise, you get the following behavior with ```mm.py README.md```:
```bash
...
Running shell 'bash -c' with command: `dapr run --app-id frontendapp --app-port 8080 --dapr-http-port 3500 node server.js`
Running shell 'bash -c' with command: `curl -w "\n" -s 'http://localhost:8080/calculate/add' -H 'Content-Type: application/json' --data '{"operandOne":"56","operandTwo":"3"}'
 curl -w "\n" -s 'http://localhost:8080/calculate/subtract' -H 'Content-Type: application/json' --data '{"operandOne":"52","operandTwo":"34"}'
 curl -w "\n" -s 'http://localhost:8080/calculate/divide' -H 'Content-Type: application/json' --data '{"operandOne":"144","operandTwo":"12"}'
 curl -w "\n" -s 'http://localhost:8080/calculate/multiply' -H 'Content-Type: application/json' --data '{"operandOne":"52","operandTwo":"34"}'
 curl -w "\n" -s 'http://localhost:8080/persist' -H 'Content-Type: application/json' --data '[{"key":"calculatorState","value":{"total":"54","next":null,"operation":null}}]'
 curl -s 'http://localhost:8080/state' | python -m json.tool`
```

This is just a convenience feature for somebody who wants to run the distributed calc demo and wants to be able to test it in their browser as well. During normal automated validation, we'll skip the pause and do everything else as normal. I'll break this into a separate step in the README.md file to make it more clear.